services/src/pro/users/hashmap_userdb.rs

```diff
@@ -16,29 +16,36 @@ use geoengine_datatypes::util::Identifier;
 
 #[derive(Default)]
 pub struct HashMapUserDb {
-    users: HashMap<String, User>,
-    sessions: HashMap<SessionId, UserSession>,
+    users: Db<HashMap<String, User>>,
+    sessions: Db<HashMap<SessionId, UserSession>>,
 }
 
 #[async_trait]
 impl UserDb for HashMapUserDb {
     /// Register a user
-    async fn register(&mut self, user_registration: Validated<UserRegistration>) -> Result<UserId> {
+    async fn register(&self, user_registration: Validated<UserRegistration>) -> Result<UserId> {
         let user_registration = user_registration.user_input;
         ensure!(
-            !self.users.contains_key(&user_registration.email),
+            !self
+                .users
+                .read()
+                .await
```

Comment:
If you need the write lock anyway, why require a read lock first and then a write lock?

Reply:
fccf47a2459c22ceb51d2ac1d1b4e0237d13cfd1